src/InOneWeekend/main.cc

```diff
@@ -113,8 +113,8 @@ int main() {
         for (int i = 0; i < image_width; ++i) {
             color pixel_color;
             for (int s = 0; s < samples_per_pixel; ++s) {
-                auto u = (i + random_double()) / image_width;
-                auto v = (j + random_double()) / image_height;
+                auto u = (i + random_double()) / (image_width-1);
```

Comment:
not off-by-1 adept atm

Reply:
?

Reply:
i may analyze this later.  you're probably correct here.

Reply:
just did a mental visualization.  if image width was 1, width-1 as a divisor would not sample correctly amirite?  think im rite

Reply:
Vectors `u` and `v` are mapped across the window to the [0,1] interval. For a 2×2 image, the old code you give you vectors from 0 to 0.5. For a 3×3 image, this would give you a vector from 0 to 0.67.

With a 1×1 image, you don't need a vector to "span" the pixel, as a single pixel is a point. The width of a 1×1 image is thus zero. The fact that every image larger than 1×1 can't have an accurate normalized (unit) width or height reinforces this.

Reply:
Related reading: _A Pixel Is Not A Little Square, A Pixel Is Not A Little Square, A Pixel Is Not A Little Square!_ → http://alvyray.com/Memos/CG/Microsoft/6_pixel.pdf